packages/eslint-plugin/tests/rules/no-array-delete.test.ts

```diff
@@ -0,0 +1,51 @@
+import rule from '../../src/rules/no-array-delete';
+import { getFixturesRootDir, RuleTester } from '../RuleTester';
+
+const rootDir = getFixturesRootDir();
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2015,
+    tsconfigRootDir: rootDir,
+    project: './tsconfig.json',
+  },
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('no-array-delete', rule, {
```

Comment:
Let's add a few more test cases to be really thorough:

* Computed values that are always a number: `arr[Math.random() ? 1 : 1]`
* Computed values that aren't always a number: `arr[Math.random() ? 1 : ""]`
* What if the element is a Sequence Expression: `arr[something(), 1]`
    * I think this is a bug in the fixer right now. Before fixing, the code would delete element `1`; after, it passes `something(), 1` as two function arguments.